functions/Read-DbaBackupHeader.ps1

```diff
@@ -90,11 +90,19 @@ Gets a list of all .bak files on the \\nas\sql share and reads the headers using
 	{
 		$LoopCnt = 1
 		$FunctionName = $FunctionName = (Get-PSCallstack)[0].Command
-		Write-Verbose "$FunctionName - Connecting to $SqlServer"
+
 		try
 		{
-			$server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $SqlCredential -ErrorVariable ConnectError
-			
+			if ($sqlServer -isnot [Microsoft.SqlServer.Management.Smo.SqlSmoObject])
+			{
+					Write-Verbose "$FunctionName - Connecting to $SqlServer"
+					$server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $SqlCredential -ErrorVariable ConnectError
+			}
+			else
+			{
+				Write-Verbose "$FunctionName - Reusing connection to $SqlServer"
```

Comment:
it seemed less than reliable in places unfortunately. I spent a week chasing my tail so decided to belt and brace stuff. 
These functions are about the only ones that can be expected to regularly work in single user mode, and as someone who's trying to restore master probably has enough on their plates I thought I'd make their life a little easier.